Index: src/main/java/net/sf/antcontrib/cpptasks/gcc/GccLinker.java
===================================================================
--- src/main/java/net/sf/antcontrib/cpptasks/gcc/GccLinker.java	(revision 177)
+++ src/main/java/net/sf/antcontrib/cpptasks/gcc/GccLinker.java	(working copy)
@@ -78,6 +78,12 @@
      */
     public String decorateLinkerOption(StringBuffer buf, String arg) {
         String decoratedArg = arg;
+        if (arg.startsWith("--sysroot")) {
+          return arg;
+        }
+        if (arg.startsWith("-nostdlib")) {
+          return arg;
+        }
         if (arg.length() > 1 && arg.charAt(0) == '-') {
             switch (arg.charAt(1)) {
                 //
